g++ 4.8 -Wmaybe-uninitialized warning in process_group_symbolizer.cpp

flagging for @zerebubuth in case this seems worth fixing. Seen on ubuntu precise.

```
src/renderer_common/process_group_symbolizer.cpp: In function ‘void mapnik::render_point_symbolizer(const mapnik::point_symbolizer&, mapnik::feature_impl&, const mapnik::proj_transform&, RendererType&, F) [with F = mapnik::render_thunk_extractor::operator()(const mapnik::point_symbolizer&) const::__lambda1; RendererType = mapnik::virtual_renderer_common]’:
src/renderer_common/process_group_symbolizer.cpp:95:71: warning: ‘*((void*)(& marker)+8).std::__shared_ptr<mapnik::marker, (__gnu_cxx::_Lock_policy)2u>::_M_ptr’ may be used uninitialized in this function [-Wmaybe-uninitialized]
             point_render_thunk thunk(pos, marker, tr, opacity, comp_op);
                                                                       ^
In file included from src/renderer_common/process_group_symbolizer.cpp:25:0:
include/mapnik/renderer_common/process_point_symbolizer.hpp:44:41: note: ‘*((void*)(& marker)+8).std::__shared_ptr<mapnik::marker, (__gnu_cxx::_Lock_policy)2u>::_M_ptr’ was declared here
     boost::optional<mapnik::marker_ptr> marker;
                                         ^
```

--- a/include/mapnik/renderer_common/process_point_symbolizer.hpp
+++ b/include/mapnik/renderer_common/process_point_symbolizer.hpp
@@ -41,15 +41,10 @@ void render_point_symbolizer(point_symbolizer const &sym,
                              F render_marker)
 {
     std::string filename = get<std::string>(sym, keys::file, feature, common.vars_);
-    boost::optional<mapnik::marker_ptr> marker;
-    if (!filename.empty())
-    {
-        marker = marker_cache::instance().find(filename, true);
-    }
-    else
-    {
-        marker.reset(std::make_shared<mapnik::marker>());
-    }
+    boost::optional<mapnik::marker_ptr> marker = filename.empty()
+       ? std::make_shared<mapnik::marker>()
+       : marker_cache::instance().find(filename, true);
+
     if (marker)
     {
         double opacity = get<double>(sym,keys::opacity,feature, common.vars_, 1.0);

